_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+SELECT 4 UNION ALL 
+SELECT 5 UNION ALL 
+SELECT 6 UNION ALL 
+SELECT 'First Contact';
+```
+```shell
+┌────────────────────┐
+│       movie        │
+│      varchar       │
+├────────────────────┤
+│ The Motion Picture │
+│ 2                  │
+│ 3                  │
+│ 4                  │
+│ 5                  │
+│ 6                  │
+│ First Contact      │
+└────────────────────┘
+```
+However, if a `UNION` type is used, each individual row retains its original data type. A `UNION` is defined using key value pairs with the key as a name and the value as the data type. This also allows the specific data types to be pulled out as individual columns:
+```sql
+CREATE TABLE movies (
+     movie UNION(numbered INT, named VARCHAR)
+);
+INSERT INTO movies 
+     VALUES ('The Motion Picture'), (2), (3), (4), (5), (6), ('First Contact');
+
+FROM movies 
+SELECT 
+     movie,
+     union_tag(movie) AS type_indicator,
+     movie.named,
+     movie.numbered;
+```
+```shell
+┌────────────────────────────────────────┬────────────────┬────────────────────┬──────────┐
+│                 movie                  │ type_indicator │       named        │ numbered │
+│ union(numbered integer, named varchar) │                │      varchar       │  int32   │
+├────────────────────────────────────────┼────────────────┼────────────────────┼──────────┤
+│ The Motion Picture                     │ named          │ The Motion Picture │     NULL │
+│ 2                                      │ numbered       │ NULL               │        2 │
+│ 3                                      │ numbered       │ NULL               │        3 │
+│ 4                                      │ numbered       │ NULL               │        4 │
+│ 5                                      │ numbered       │ NULL               │        5 │
+│ 6                                      │ numbered       │ NULL               │        6 │
+│ First Contact                          │ named          │ First Contact      │     NULL │
+└────────────────────────────────────────┴────────────────┴────────────────────┴──────────┘
+```
+
+## Additional friendly features
+
+Several other friendly features are worth mentioning and some are powerful enough to warrant their own blog posts. 
+
+DuckDB takes a nod from the [`describe` function in Pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.describe.html) and implements a `SUMMARIZE` keyword that will calculate a variety of statistics about each column in a dataset for a quick, high-level overview. Simply prepend `SUMMARIZE` to any table or `SELECT` statement. 
+
+Have a look at the [correlated subqueries post](https://duckdb.org/2023/05/26/correlated-subqueries-in-sql.html) to see how to use subqueries that refer to each others’ columns. DuckDB’s advanced optimizer completely solves performance problems when expressing a query as naturally as possible. What was once an anti-pattern for performance reasons can now be used freely!
```

Comment:
"DuckDB’s advanced optimizer completely solves performance problems"
I think that may be an overstatement, there are still many challenges for query optimization and queries sometimes need to be rewritten for better performance. Our query optimizer solves many common issues (like correlated subqueries, detecting anti-/semijoins), so the statement should be fine if it is scoped down a bit.

Reply:
I rephrased it completely to something more accurate! That was some hyperbole, you are correct.